recommender/engines/persistent_mixin.py

```diff
@@ -17,35 +15,49 @@ class Persistent(ABC):
     """
 
     @classmethod
-    def load(cls, version: str = None) -> Any:
+    def check_existence(cls, version: str = None) -> Any:
         """
-        Load object from MLComponents. It will look for object with type of the
-         class that inherit this mixin and with provided version. It will
-         return the newest one object in the DB that matches criteria.
+        Check whether certain ML model exists in the MLComponents
+         and return the newest object from the DB that matches criteria.
         Args:
             version: Any, user-defined string.
 
         Returns:
-            ml_component: Newest object that matches criteria.
-
+            ml_component: Newest ML model that matches criteria.
         """
         if version is None:
-            last_object = MLComponent.objects(type=cls.__name__).order_by("-id").first()
+            ml_component = (
+                MLComponent.objects(type=cls.__name__).order_by("-id").first()
+            )
         else:
-            last_object = (
+            ml_component = (
                 MLComponent.objects(type=cls.__name__, version=version)
                 .order_by("-id")
```

Comment:
just noticed, we shouldn't rely on the order of ids. We should have a "created" column and sort by it - Please move it to another issue. We have to address it,  because it just might be application-breaking

Reply:
I agree we should include the date while saving ML models, and updating current models accordingly.
I made a ticket https://github.com/cyfronet-fid/recommender-system/issues/372